### PR TITLE
Relayer Fee and Refund Support

### PR DESCRIPTION
**Pull request by AtelyPham**
**Note:** This PR needs to be tested in the live version with a relayer that supports fees and refunds. Alternatively, it can be tested using a local relayer. @salman01zp 

## Summary of changes
_Provide a detailed description of proposed changes._
- Adding relayer fee in withdraw flow
- Refund support in withdraw flow

### Proposed area of change
_Put an `x` in the boxes that apply._

- [x] `apps/bridge-dapp`
- [ ] `apps/stats-dapp`
- [ ] `apps/webbsite`
- [x] `libs/webb-ui-components`

### Reference issue to close (if applicable)
_Specify any issues that can be closed from these changes (e.g. Closes #233)._
- Closes https://github.com/webb-tools/webb-dapp/issues/1039

### Screen Recording
_If possible provide a screen recording of proposed change._

https://user-images.githubusercontent.com/60747384/224765492-ffff1721-2b14-4bc0-b8d5-1eea4dc1dd9a.mp4

### CODE REVIEW

**Comment by drewstone** [comment]
@AtelyPham the MAX refund selection must be taken directly from the relayer's information not from the user's available balance.

The amount received is the `amount + refund - (fee)`

**Comment by AtelyPham** [comment]
Updated based on feedback. Please give it a review whenever you can @shekohex @drewstone.

https://user-images.githubusercontent.com/60747384/225092012-3a695a48-6c08-4e33-bda2-6102674dbea4.mp4

**Review by shekohex — COMMENTED** [review]
LGTM! 

<small>_from reviewing the demo video_</small>

**Comment by dutterbutter** [comment]
@AtelyPham We need to add more details to the fee description here:

CleanShot 2023-03-16 at 13 43 55

I think this could be become quite confusing to new users. A proposed solution is simply to break up to be more explicit. So:

```
Withdrawing: .956 webbWETH
Receiving refund amount: .001 ETH
Remaining balance: 7 webbWETH
Transaction fee: .043 webbWETH
Relayer refund fee: .001 webbWETH 
```

Then in confirmation card:

We currently have this:
CleanShot 2023-03-16 at 13 49 09

**Proposed breakdown:**

```
Withdrawing: .956 webbWETH
Receiving refund amount: .001 ETH
Remaining balance: 7 webbWETH
Transaction fee: .043 webbWETH
Relayer refund fee: .001 webbWETH 
```

Lastly, I do not think we should include the .001 ETH here. I think it could be misleading as we are not withdrawing .001 ETH from the shielded pool rather we are receiving .001 ETH from the relayer. I propose removing .001 ETH from this line item.

CleanShot 2023-03-16 at 13 50 56

**Comment by AtelyPham** [comment]
Follow your suggestion, but I think, the withdrawing should be the total amount. Something like this is more clear to me

```
Withdrawing: 1 webbWETH
Receiving: .956 webbWETH
Receiving refund amount: .001 ETH
Remaining balance: 7 webbWETH
Transaction fee: .043 webbWETH
Relayer refund fee: .001 webbWETH 
```

Wdyt? @dutterbutter

**Comment by AtelyPham** [comment]
Actually, we can add the tooltip to provide more info about these numbers. If you think we should do that, just send me the content to explain these numbers, I can add tooltips. @dutterbutter

**Comment by AtelyPham** [comment]
Resolved comments.

https://user-images.githubusercontent.com/60747384/227207703-831d977c-3b7a-4da9-8f12-a7a1e8d5fc60.mp4

**Comment by dutterbutter** [comment]
@AtelyPham can you change:

"Receiving Refund Amount" to "Refund Amount"

**Comment by AtelyPham** [comment]
Getting the fee info when no relayer is selected and updating to new deployments.

https://user-images.githubusercontent.com/60747384/227357259-be1e18b1-6854-4618-8687-fce7411e5ab4.mp4

File: libs/dapp-config/src/anchors/anchor-config.ts [hunk 1]
```diff
@@ -6,22 +6,25 @@ import { CurrencyConfig } from '../currencies';
...
+    '0x38e7aa90c77f86747fab355eecaa0c2e4c3a463d': 3146553,
   },
   [calculateTypedChainId(ChainType.EVM, EVMChainId.PolygonTestnet)]: {
-    '0xa1a2b7e08793b3033122b83cbee56726678588b5': 32139400,
+    '0x38e7aa90c77f86747fab355eecaa0c2e4c3a463d': 33462722,
   },
   [calculateTypedChainId(ChainType.EVM, EVMChainId.MoonbaseAlpha)]: {
-    '0xa1a2b7e08793b3033122b83cbee56726678588b5': 3771120,
+    '0x38e7aa90c77f86747fab355eecaa0c2e4c3a463d': 3996742,
   },
```
Comment (dutterbutter) [inline libs/dapp-config/src/anchors/anchor-config.ts:28]: @AtelyPham I do not want to replace the bridge existing bridge deployments (e.g. `webbWETH`) but simply to add `webbAlpha`. We are adding a bridge not replacing one.

Comment (dutterbutter) [reply libs/dapp-config/src/anchors/anchor-config.ts]: I see I see.

**Comment by dutterbutter** [comment]
@AtelyPham we need to also include the fees in the deposit, transfer flows. Did you want to include that in this PR or in a subsequent one?

**Comment by AtelyPham** [comment]
> @AtelyPham we need to also include the fees in the deposit, transfer flows. Did you want to include that in this PR or in a subsequent one?

Let's do in subsequent PR, because we need to create new logic for that.

**Comment by dutterbutter** [comment]
This LGTM but we will need to merge #1048 prior to this one to handle the bridge selection logic.

**Review by dutterbutter — APPROVED** [review]

**Pull request merged**